### ISSUE TITLE
ar-hit-test enabled off makes the entity "lose track"

### ISSUE THREAD

**Issue by Mrbside** [aframevr/aframe#5324]
**Description:**

- A-Frame Version: 1.4.2
- Platform / Device: Android Moto g100
- Reproducible Code Snippet or URL: https://glitch.com/edit/#!/exciting-thankful-microwave

Turning enabled : false in the ar-hit-test component makes the entities lose track

Sorted this out by instead changing the target and src parameter after successfully placing the object on the ar surface.

Should the enabled parameter work?

Extra bug: in the inspector when I click on the entities called badger and car the UI disappears

**Comment by dmarcos** [comment]
Can you provide link to simple example to reproduce? https://glitch.com/~aframe is great. Thanks

**Comment by Mrbside** [comment]
Sure thing 

https://glitch.com/edit/#!/exciting-thankful-microwave

Here you scan the floor, see a red reticle, tap on the screen and my object is spawned on the floor, in the "ar-hit-test-select" event, I turn enabled off in the ar-hit-test, If i do it immediately you dont see the object appear, so I added a timeout of 2 seconds to do it, after that you can see the object "disappear" but its far off the camera at your right....}


Also I've seen people report problems with the inspector, UI disappearing when some entity is clicked, here happens the same, try click on the entity called car or badger and this bug happens, also happening if I click on other entity and go to badger or car using arrow keys.

**Comment by Mrbside** [comment]
> Can you provide link to simple example to reproduce? https://glitch.com/~aframe is great. Thanks
 
Sure thing

https://glitch.com/edit/#!/exciting-thankful-microwave

Here you scan the floor, see a red reticle, tap on the screen and my object is spawned on the floor, in the "ar-hit-test-select" event I turn enabled false in the ar-hit-test, If i do it immediately you dont see the object appear, so I added a timeout of 2 seconds to do it, after that you can see the object "disappear" but its far off the camera at your left....}

Also I've seen people report problems with the inspector, UI disappearing when some entity is clicked, here happens the same, try click on the entity called car or badger and this bug happens, also happening if I click on other entity and go to badger or car using arrow keys.

**Comment by Mrbside** [comment]
> > Extra bug: in the inspector when I click on the entities called badger and car the UI disappears
> 
> The url used to download the aframe-inspector is https://unpkg.com/aframe-inspector@1.4.x/dist/aframe-inspector.min.js that resolves to 1.4.1. @dmarcos It seems you didn't release aframe-inspector 1.4.2? master have some fixes related to the panels disappearing. #5241

So could it temporary fix this by replacing the url on the inspector property on a-scene? What URL should I use?

**Comment by Mrbside** [comment]
Sorry accidentally closed this 😅

**Comment by Mrbside** [comment]
Regarding the hit-test component, is there any fix?

**Comment by vincentfretin** [comment]
@Mrbside Yes you can create a build of aframe-inspector, see my comment https://github.com/aframevr/aframe/issues/5241#issuecomment-1498664170 If you still have an issue with latest version, please create an issue in aframe-inspector repo with a simple glitch reproducing the issue.

**Comment by Mrbside** [comment]
> @Mrbside Yes you can create a build of aframe-inspector, see my comment [#5241 (comment)](https://github.com/aframevr/aframe/issues/5241#issuecomment-1498664170) If you still have an issue with latest version, please create an issue in aframe-inspector repo with a simple glitch reproducing the issue.

Everything great now that dmarcos updated the inspector, the issue is gone using 1.4.2  :)

**Comment by dmarcos** [comment]
@Mrbside Can we close the issue?

**Comment by Mrbside** [comment]
> @Mrbside Can we close the issue?

Oh well actually the main issue of this thread is about the hit test component...

The inspector stuff was a secondary observation 😅

**Comment by Mrbside** [comment]
@dmarcos @vincentfretin Hey guys I'm having again the problem with the editor UI disappearing, I'm using vanilla aframe 1.4.2, no AR library
<script src="https://aframe.io/releases/1.4.2/aframe.min.js"></script>
Here are screenshots of the console and the entity it refers 

Thanks 
![image](https://github.com/aframevr/aframe/assets/9389898/880eb58a-05b3-48cb-84d9-f827aea780d9)
![image](https://github.com/aframevr/aframe/assets/9389898/a1a19f42-a880-4f93-9fe2-444b18644e7f)
![image](https://github.com/aframevr/aframe/assets/9389898/fcbd05ed-d2eb-4871-931a-ae3f71605fd8)

**Comment by Mrbside** [comment]
I tried using the master, but when I do this my occlusion stops working for some reason, no errors on console.

This is my occlusion component that works on 1.4.2 but not on master...

AFRAME.registerComponent('occlusion-material', {
    init: function() {
        this.el.addEventListener('model-loaded', e=>{
            let mesh = this.el.getObject3D('mesh')
            if (mesh !== undefined) {
                mesh.traverse(function(node) {
                if (node.isMesh) {
                    if (node.material){
                        console.log(node)
                        var material = new THREE.MeshBasicMaterial( {colorWrite: false} );
                        node.material = material;
                        node.needsupdate = true;
                        node.frustumCulled = false;
                    }
                }
                });
            }
        });
    }
}); 
![Captura de pantalla 2023-10-06 141522](https://github.com/aframevr/aframe/assets/9389898/108c0e9d-449e-4871-84ec-dccf01073889)
![Captura de pantalla 2023-10-06 141543](https://github.com/aframevr/aframe/assets/9389898/18be9a15-ff01-4b13-aa32-6b5ec1ec2f97)

**Comment by Mrbside** [comment]
And anyways the master has the same inspector bug 😅 

@dmarcos @vincentfretin thank you in advance

**Comment by mrxz** [comment]
> I tried using the master, but when I do this my occlusion stops working for some reason, no errors on console, im using gltfs

This might be due to objects now being sorted by default on master (see #5341). Since you want your occluders to be rendered first you can set their [renderOrder](https://threejs.org/docs/index.html#api/en/core/Object3D.renderOrder) to something like -1.

**Comment by vincentfretin** [comment]
@Mrbside That's a new stacktrace we never saw here. Please create a new issue in https://github.com/aframevr/aframe-inspector/ with a link and steps to reproduce.

**Comment by vincentfretin** [comment]
@Mrbside Actually this is not an issue in aframe-inspector. In html, ids can't start with a number, you need to change your id.
